client/my-sites/posts/post-controls.jsx

```diff
@@ -1,244 +1,223 @@
-
-			for ( i = moreControlsSpliceIndex; i < availableControls.length; i++ ) {
-				extraControls.push( availableControls[i] );
-			}
-
-			availableControls.splice( moreControlsSpliceIndex );
-		}
-
-		// --- Return Controls ---
-		postControlsClass = 'post-controls';
-
-		if ( extraControls.length ) {
-			availableControls.push( {
-				text: this.translate( 'More' ),
-				className: 'post-controls__more',
-				onClick: this.props.onShowMore,
-				icon: 'ellipsis'
-			} );
-
-			extraControls.push( {
-				text: this.translate( 'Back' ),
-				className: 'post-controls__back',
-				onClick: this.props.onHideMore,
-				icon: 'chevron-left'
+	} else if ( userCan( 'delete_post', post ) ) {
+		controls.main.push( {
+			className: 'restore',
+			icon: 'undo',
+			onClick: onRestore,
+			text: translate( 'Restore' ),
+		} );
+	}
+
+	if ( userCan( 'delete_post', post ) ) {
+		if ( 'trash' === post.status ) {
+			controls.main.push( {
+				className: 'trash is-scary',
+				icon: 'trash',
+				onClick: onDelete,
+				text: translate( 'Delete Permanently' ),
 			} );
-
-			mainControls = this.buildControls( availableControls );
-			moreControls = this.buildControls( extraControls );
-			postControlsClass += ( extraControls.length <= 2 ) ? ' post-controls--desk-nomore' : '';
-
-			return (
-				<div className={ postControlsClass }>
-					<ul className="post-controls__pane post-controls__more-options">
-						{ moreControls }
-					</ul>
-					<ul className="post-controls__pane post-controls__main-options">
-						{ mainControls }
-					</ul>
-				</div>
-			);
 		} else {
-			mainControls = this.buildControls( availableControls );
-
-			return (
-				<div className={ postControlsClass }>
-					<ul className="post-controls__pane post-controls__main-options">
-						{ mainControls }
-					</ul>
-				</div>
-			);
+			controls.main.push( {
+				className: 'trash',
+				icon: 'trash',
+				onClick: onTrash,
+				text: translate( 'Trash' ),
+			} );
 		}
-	},
-
-} );
+	}
+
+	if ( ( 'publish' === post.status || 'private' === post.status ) && userCan( 'edit_post', post ) ) {
+		controls.main.push( {
+			className: 'copy',
+			href: `/post/${ site.slug }?copy=${ post.ID }`,
+			icon: 'clipboard',
+			onClick: copy,
+			text: translate( 'Copy' ),
+		} );
+	}
+
+	// More Controls (behind ... more link)
+	if ( controls.main.length > 4 && fullWidth ) {
+		controls.more = controls.main.splice( 4 );
+	} else if ( controls.main.length > 2 && ! fullWidth ) {
+		controls.more = controls.main.splice( 2 );
+	}
+
+	if ( controls.more.length ) {
+		controls.main.push( {
+			className: 'more',
+			icon: 'ellipsis',
+			onClick: onShowMore,
+			text: translate( 'More' ),
+		} );
+
+		controls.more.push( {
+			className: 'back',
+			icon: 'chevron-left',
+			onClick: onHideMore,
+			text: translate( 'Back' ),
+		} );
+	}
+
+	return controls;
+};
+
+const getControlElements = controls => controls.map( ( control, index ) =>
```

Comment:
If we want this to exist as it's own component, please split this out into it's own file. Otherwise I'd recommend `extending Component` instead with helper render methods